src/machine_charm.py

```diff
@@ -255,7 +255,8 @@ def metrics_jobs(self) -> list:
         for job in jobs:
             static_configs = job.get("static_configs", [])
             for static_config in static_configs:
-                static_config["labels"] = self._principal_labels
+                static_config.setdefault("labels", {})
+                static_config["labels"].update(**self._principal_labels)
```

Comment:
TIL about `setdefault`, but do you think this is more straightforward?
```suggestion
                static_config["labels"] = {
                    # Be sure to keep labels from static_config
                    **static_config.get("labels", {}),
                    **self._principal_labels,
                }
```

Reply:
I had the foggiest memory that `{**dict1, **dict2}` breaks if there is a common set of keys, but apparently this does not seem to be the case:

```
>>> a = {"a": "b"}
>>> b = {"a": "c"}
>>> {**a, **b}
{'a': 'c'}
```

Sure, no problem to do it like this :+1: